src/allmydata/scripts/tahoe_ls.py

```diff
@@ -13,184 +13,199 @@ def list(options):
+                classify = ""
+                if rw_uri:
+                    classify = "*"
+            else:
+                has_unknowns = True
+                t0 = "?"
+                size = "?"
+                classify = "?"
+            t1 = "-"
+            if ro_uri:
+                t1 = "r"
+            t2 = "-"
+            if rw_uri:
+                t2 = "w"
+            t3 = "-"
+            if childtype == "dirnode":
+                t3 = "x"
+
+            uri = rw_uri or ro_uri
+
+            line = []
+            if options["long"]:
+                line.append(t0 + t1 + t2 + t3)
+                line.append(size)
+                line.append(ctime_s)
+            if not options["classify"]:
+                classify = ""
+
+            encoding_error = False
+            try:
+                line.append(unicode_to_output(name) + classify)
+            except UnicodeEncodeError:
+                encoding_error = True
+                line.append(quote_output(name) + classify)
+
+            if options["uri"]:
+                line.append(uri)
+            if options["readonly-uri"]:
+                line.append(quote_output(ro_uri or "-", quotemarks=False))
+
+            rows.append((encoding_error, line))
+
+        max_widths = []
+        left_justifys = []
+        for (encoding_error, row) in rows:
+            for i, cell in enumerate(row):
+                while len(max_widths) <= i:
+                    max_widths.append(0)
+                while len(left_justifys) <= i:
+                    left_justifys.append(False)
+                max_widths[i] = max(max_widths[i], len(cell))
+                if cell.startswith("URI"):
+                    left_justifys[i] = True
+        if len(left_justifys) == 1:
+            left_justifys[0] = True
+        fmt_pieces = []
+        for i in range(len(max_widths)):
+            piece = "%"
+            if left_justifys[i]:
+                piece += "-"
+            piece += str(max_widths[i])
+            piece += "s"
+            fmt_pieces.append(piece)
+        fmt = " ".join(fmt_pieces)
+
+        for (encoding_error, row) in rows:
+            if encoding_error:
+                print((fmt % tuple(row)).rstrip(), file=stderr)
+                rc = 1
+            else:
+                print((fmt % tuple(row)).rstrip(), file=stdout)
+
+        if rc == 1:
+            print("\nThis listing included files whose names could not be converted to the terminal" \
+                  "\noutput encoding. Their names are shown using backslash escapes and in quotes.", file=stderr)
+        if has_unknowns:
+            print("\nThis listing included unknown objects. Using a webapi server that supports" \
+                  "\na later version of Tahoe may help.", file=stderr)
+        return rc
+
+    rc = []
+    if not where:
+        rc.append(list_dir(u''))
+    else:
+        for directory in where:
+            rc.append(list_dir(directory))
+    return min(rc)
```

Comment:
return success if at least one operation was a success.

Reply:
Might be better to have the above comment in the code rather than a long-lost PR